contracts/GuestList.vy

```diff
@@ -43,17 +55,47 @@ def set_bouncer(new_bouncer: address):
     self.bouncer = new_bouncer
 
 
+@view
+@internal
+def _total_yfi(user: address) -> uint256:
+    return (
+        self.yfi.balanceOf(user)
+        + self.ygov.balanceOf(user)
+        + self.yyfi.balanceOf(user) * self.yyfi.getPricePerFullShare() / 10 ** 18
+    )
+
+
+@view
+@internal
+def _time_factor(bag: uint256) -> uint256:
+    if block.timestamp > self.activation + self.ape_out:
+        return 0
+    return bag - bag * (block.timestamp - self.activation) / self.ape_out
+
+
+@view
+@external
+def total_yfi(user: address) -> uint256:
+    """
+    Total YFI in wallet + yGov + YFI vault.
+    """
+    return self._total_yfi(user)
+
+
+@view
+@external
+def entrance_cost() -> uint256:
+    return self._time_factor(self.min_bag)
+
+
 @view
 @external
-def authorized(guest: address, amount: uint256) -> bool:
+def authorized(guest: address, amount: uint256 = 0) -> bool:
     """
     Check if a user with a bag of certain size is allowed to the party.
     """
     if self.guests[guest]:
+        return True    
+    if block.timestamp > self.activation + self.ape_out:
         return True
```

Comment:
should already be covered with below logic, no? (albeit more gas)

Reply:
Yes, changed that